source/backend/dal/Repositories/BasePimsRepository.cs

```diff
@@ -0,0 +1,75 @@
+using System.Security.Claims;
+using Microsoft.EntityFrameworkCore.Storage;
+using Microsoft.Extensions.Logging;
+
+namespace Pims.Dal.Repositories
+{
+    /// <summary>
+    /// BaseService abstract class, provides a generic service layer to perform CRUD operations on the datasource.
+    /// </summary>
+    public abstract class BasePimsRepository : IRepository
+    {
+        #region Variables
```

Comment:
I thought we already had a "base" repository class to provide common functionality. Why create a new one? If more common functions are needed, why not implement them in the existing base class?

![image](https://github.com/user-attachments/assets/580a0abb-73cb-40c8-82fd-63c225709e35)